contracts/SeedsOnEarth.sol

```diff
@@ -83,23 +85,36 @@ contract SeedsOnEarth {
 
         quests.push(quest);
 
-        emit SponserQuest(quests.length - 1, _tokenAddress, quest.amount, _description);
+        emit SponserQuest(quests.length - 1, _tokenAddress, quest.amount, _ipfsHash);
     }
-    
+
+    /**
+    * @notice join a quest, called by user to join a quest
+    * @param _questId id of quest
+    **/
+    function joinQuest(uint256 _questId) public {
+        Quest storage quest = quests[_questId];
+        require(quest.status == QuestStatus.PENDING, "Quest must be pending to join");
+        quest.users.push(msg.sender);
+        usersQuestsMapping[msg.sender].push(_questId);
+        if (quest.users.length == quest.numOfUsers) {
```

Comment:
this will always be true

Reply:
Understood, Liron.

Assigning dynamic size array throws this error:
```
Invalid type for argument in function call. Invalid implicit conversion from function (uint256) pure returns (address[] memory) to address[] memory requested.
```

Should I create a separate counter for number of users?

Reply:
only problem is that now if the quest requires only one user, the user will have to "join" and then "pickup"

Reply:
and then can keep the quest stuck with just joining.. but that's ok for now i guess

Reply:
That's right.
I think I can handle that in frontend for now. If `numOfUsers` is 1, then I will run `pickupQuest` right after `joinQuest`